File: push/src/main/java/com/nihility/service/MessengerAbility.java
package com.nihility.service;

import com.xiaomi.push.service.XMPushServiceMessenger;

public class MessengerAbility implements XMPushServiceListener {
    private final XMPushServiceMessenger messenger;

    public MessengerAbility(XMPushServiceMessenger messenger) {
        this.messenger = messenger;
    }

    @Override
    public void connectionStatusChanged(ConnectionStatus connectionStatus) {
        messenger.notifyConnectionStatusChanged(connectionStatus.ordinal());
    }
}

File: push/src/main/java/com/nihility/service/MessengerAbility.kt
package com.nihility.service

import com.xiaomi.push.service.XMPushServiceMessenger

class MessengerAbility(
    private val messenger: XMPushServiceMessenger
) : XMPushServiceListener {
    override fun connectionStatusChanged(connectionStatus: XMPushServiceListener.ConnectionStatus) {
        messenger.notifyConnectionStatusChanged(connectionStatus.ordinal)
    }
}
